entrypoint.sh

```diff
@@ -19,29 +20,68 @@ check_database_connection() {
     timeout=$(( $timeout - 1 ))
     if [[ $timeout -eq 0 ]]; then
       echo
-      echo "Could not connect to database server. Aborting..."
-      return 1
+      echo "Could not connect to database server! Aborting..."
+      exit 1
     fi
     echo -n "."
     sleep 1
   done
   echo
 }
 
+checkdbinitmysql() {
+    table=sessions
+    if [ $(mysql -N -s -h ${DB_HOST} -u ${DB_USERNAME} ${DB_PASSWORD:+-p$DB_PASSWORD} -e \
+        "select count(*) from information_schema.tables where \
+            table_schema='${DB_DATABASE}' and table_name='${table}';") -eq 1 ]; then
+        echo "Table ${table} exists! ..."
+    else
+        echo "Table ${table} does not exist! ..."
+        init_db
+    fi
+
+}
+
+checkdbinitpsql() {
+    table=sessions
+    export PGPASSWORD=${DB_PASSWORD}
+    if [ "$(psql -h ${DB_HOST} -U ${DB_USERNAME} -c "SELECT to_regclass('${table}');" | grep -c "${table}")" -eq 1 ]; then
+        echo "Table ${table} exists! ..."
+    else
+        echo "Table ${table} does not exist! ..."
+        init_db
+    fi
+
+}
+
+check_configured() {
+  case ${DB_DRIVER} in
+    mysql)
+      checkdbinitmysql
+      ;;
+    pgsql)
+      checkdbinitpsql
+      ;;
+  esac
+}
+
 initialize_system() {
+  echo "Initializing Cachet container ..."
   APP_ENV=${APP_ENV:-development}
   APP_DEBUG=${APP_DEBUG:-true}
   APP_URL=${APP_URL:-http://localhost}
-  APP_KEY=${APP_KEY:-SECRET}
+  APP_KEY=${APP_KEY:-base64:SGZXUdds0Qnbf55/7diaHMPPM2TXfOSxHtUAXz6POSw=}
```

Comment:
I realize a key is being shipped here, but it needed some default value for now.